components/esp_modem/examples/pppos_client/main/pppos_client_main.c

```diff
@@ -134,17 +163,6 @@ void app_main(void)
     dte_config.task_priority = CONFIG_EXAMPLE_MODEM_UART_EVENT_TASK_PRIORITY;
     dte_config.dte_buffer_size = CONFIG_EXAMPLE_MODEM_UART_RX_BUFFER_SIZE / 2;
 
-    /* Configure the DCE */
-    esp_modem_dce_config_t dce_config = ESP_MODEM_DCE_DEFAULT_CONFIG(CONFIG_EXAMPLE_MODEM_PPP_APN);
-
-    /* Configure the PPP netif */
-    esp_netif_config_t netif_ppp_config = ESP_NETIF_DEFAULT_PPP();
```

Comment:
Here I wanted to demonstrate the sequence of config and init, so configure the DTE, then DCE and then PPP. And after that we would start creating objects (in C we do that in one step). I'm not sure if we could keep this order, maybe if we make the UART part destroyable, too?

Reply:
Yes, the UART and netif objects should be destroyed and then recreated. But I didn't want to make this PR too big. And since you already have an idea about doing this, I'd prefer doing it in follow-up PR.